x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/X-RateGroupingAggregatorFunction.java.st

```diff
@@ -654,12 +662,12 @@ public final class Rate$Type$GroupingAggregatorFunction implements GroupingAggre
      * We still extrapolate the rate in this case, but not all the way to the boundary, only by half of the average duration between
      * samples (which is our guess for where the series actually starts or ends).
      */
-    private static double extrapolateRate(ReducedState state, long rangeStart, long rangeEnd) {
+    private static double extrapolateRate(ReducedState state, long rangeStart, long rangeEnd, RateCalculationMode calculation) {
         assert state.samples >= 2 : "rate requires at least two samples; got " + state.samples;
         final long firstTS = state.intervals[state.intervals.length - 1].t2;
         final long lastTS = state.intervals[0].t1;
         double firstValue = state.intervals[state.intervals.length - 1].v2;
-        double lastValue = state.intervals[0].v1 + state.resets;
+        double lastValue = (calculation == RateCalculationMode.DELTA) ? state.intervals[0].v1 : state.intervals[0].v1 + state.resets;
```

Comment:
Since we don't need to track resets, we should have a separate implementation for delta(), as tracking resets is too expensive.